libs/datasets/dataset_orchestrator.py

```diff
@@ -0,0 +1,319 @@
+                multiregion_dataset
+            )
+            if self.print_stats:
+                multiregion_dataset.print_stats("patch_maryland_missing_case_data")
+
+        multiregion_dataset = nytimes_anomalies.filter_by_nyt_anomalies(multiregion_dataset)
+        if self.print_stats:
+            multiregion_dataset.print_stats("nytimes_anomalies")
+
+        multiregion_dataset = outlier_detection.drop_new_case_outliers(multiregion_dataset)
+        multiregion_dataset = outlier_detection.drop_new_deaths_outliers(multiregion_dataset)
+        if self.print_stats:
+            multiregion_dataset.print_stats("outlier_detection")
+
+        multiregion_dataset = timeseries.drop_regions_without_population(
+            multiregion_dataset, KNOWN_LOCATION_ID_WITHOUT_POPULATION, _log
+        )
+        if self.print_stats:
+            multiregion_dataset.print_stats("drop_regions_without_population")
+
+        if state.state == "PR":
+            multiregion_dataset = custom_aggregations.aggregate_puerto_rico_from_counties(
+                multiregion_dataset
+            )
+            if self.print_stats:
+                multiregion_dataset.print_stats("aggregate_puerto_rico_from_counties")
+
+        if state.state == "NY":
+            multiregion_dataset = custom_aggregations.aggregate_to_new_york_city(
+                multiregion_dataset
+            )
+            if self.print_stats:
+                multiregion_dataset.print_stats("aggregate_to_new_york_city")
+
+        hsa_aggregator = statistical_areas.CountyToHSAAggregator.from_local_data()
+        multiregion_dataset = hsa_aggregator.aggregate(
+            multiregion_dataset, restrict_to_current_state=state
+        )
+        if self.print_stats:
+            multiregion_dataset.print_stats("CountyToHSAAggregator")
+
+        if state.state == "DC":
+            multiregion_dataset = custom_aggregations.replace_dc_county_with_state_data(
+                multiregion_dataset
+            )
+            if self.print_stats:
+                multiregion_dataset.print_stats("replace_dc_county_with_state_data")
+        return multiregion_dataset
+
+
+def load_bulk_dataset(refresh_datasets: Optional[bool] = True) -> MultiRegionDataset:
+    if refresh_datasets:
+        timeseries_field_datasets = load_datasets_by_field(
+            combined_datasets.ALL_TIMESERIES_FEATURE_DEFINITION
+        )
+        static_field_datasets = load_datasets_by_field(
+            combined_datasets.ALL_FIELDS_FEATURE_DEFINITION
+        )
+
+        multiregion_dataset = timeseries.combined_datasets(
+            timeseries_field_datasets, static_field_datasets
+        )
+        multiregion_dataset.to_compressed_pickle(dataset_utils.COMBINED_RAW_PICKLE_GZ_PATH)
+    else:
+        multiregion_dataset = timeseries.MultiRegionDataset.from_compressed_pickle(
+            dataset_utils.COMBINED_RAW_PICKLE_GZ_PATH
+        )
+    return multiregion_dataset
+
+
+def load_datasets_by_field(
```

Comment:
There's probably work that could be done to speed up the dataset loading process (which takes ~15-20 min atm), but I didn't look into that in this PR